sympy/physics/units/dimensions.py

```diff
@@ -572,3 +572,49 @@ def is_consistent(self):
         # dimensions
         # in vector language: the set of vectors do not form a basis
         return self.inv_can_transf_matrix.is_square
+    def get_dimensionless_numbers(self, list_of_derived_quantities):
+        """ list_of_derived_quantities is the input list of derived quantities in the form of a python dictionary of tuples (key, item) where key is the name of
+        the physical quantity and item is an instance of Dimension class which possesses the dimensions of the quantity. This function is designed to take in
+        a set of physical variables (all essentially dimensionful) and returns one set of all possible dimensionless quantities, as guided by the
+        Buckingham's pi theorem (a generalization of Rayleigh's method of dimensional analysis .....) Please refer and run to the unit testing file
+        /units/test/test_dimensionless.py for the format of the input .... As a heads up, we first need to define a global set of base dimensions and with
+        those we can define the derived quantities ..... """
+        exponent_matrix = []
+        # We extract the exponent matrix from the dimensional dependencies of the quantities
+        for quantity in list_of_derived_quantities:
+            element_for_quantity = [0 for k in range(len(self.base_dims))]
+            dimensions = self.get_dimensional_dependencies(list_of_derived_quantities[quantity])
+            for base_dims in dimensions:
+                element_for_quantity[self.base_dims.index(base_dims)] = dimensions[base_dims]
+            exponent_matrix.append(element_for_quantity)
+        # Next we obtain the null space of the exponent matrix
+        exponent_matrix = Matrix(exponent_matrix)
+        exponents = exponent_matrix.T.nullspace()
+        # We need to make sure that we obtain integer nullspaces because
+        # Dimension() is not able to simplify non-diophantine exponents
+        null_space_dims = len(exponents)
+        m = 1
+        while True:
+            flag = True
+            for k in range(null_space_dims):
+                for l in range(len(exponents[k])):
+                    exponents[k][l] = exponents[k][l] * m
+                    flag = flag and (exponents[k][l]-int(exponents[k][l])) == 0.0
+            if flag:
+                # Just to make sure that the exponents are all integers
+                for k in range(null_space_dims):
+                    for l in range(len(exponents[k])):
+                        exponents[k][l] = int(exponents[k][l])
```

Comment:
This flag block looks pretty useless, why not just store integers in the matrix at its creation?